pkg/endpointmanager/subscribe.go

```diff
@@ -10,12 +10,16 @@ type Subscriber interface {
 	// EndpointCreated is called at the end of endpoint creation.
 	// Implementations must not attempt write operations on the
```

Comment:
Could add the caveat of not locking the endpoint manager at all, since this suggests it's OK to do read operations, which it clearly isn't